src/libraries/Microsoft.PowerFx.Core/Entities/External/IExternalOptionSet.cs

```diff
@@ -20,8 +20,77 @@ internal interface IExternalOptionSet : IExternalEntity
+        //          Concatenation with the backing kind is not allowed (see below for this case).
+        //          Used by JSONFormat, MatchOptions, ...
+        // 
+        // Option set is shorthand for the backind kind: CanCoerceToBackingKind = true. The option set is a convenient
+        //          shorthand for using the backind kind. Loosely typed, the only type safety is that a function defined
+        //          with the option set can't be passed the backing type.
+        //          Used by Color.
+        //
+        // Option set is shorthand for the backind kind, with Numerical Compare: CanCoreceToBackindKind and
+        //          CanCompareNumerical = true, number backed only. Numerical order comparisons (<, >, <=, >=)
+        //          are supported.
+        //          Used by ErrorKind.
+        //
+        // Custom backing kind can be used in place of the option set:
+        //          CanCoerceFromBackindKind = true, not supported for Color.
+        //          Functions that have this option set as a parameter type can be passed the backing kind.
+        //          Used by DateTimeFormat.
+        //
+        // Custom backing kind mixes with the option set:
+        //          CanCoerceFromBackindKind and CanContenateStronglyTyped = true.
+        //          Functions that have this option set as a parameter type can be passed the backing kind and
+        //          concatenations between the option set and the backind kind result in the option set type.
+        //          Used by Match.
+
+        /// <summary>
+        /// Backing kind typed values can be used in placed of this option set value.  
+        /// This is also possible for concatenate and numeric compare if those flags are set.
+        /// Examples: a completely custom regular expression string can passed for a Match enum
+        /// and a number can be used in palce of an ErrorKind.
+        /// </summary>
+        bool CanCoerceFromBackingKind { get; }
+
+        /// <summary>
+        /// Option set can be used in place of the backing kind. 
+        /// This is useful to provide a convenient enum value for the backing kind, for example ErrorKind and Color.
+        /// There is no type safety, effectively the enum is just like using the backing type.
+        /// </summary>
+        bool CanCoerceToBackingKind { get; }
+
+        /// <summary>
+        /// Only applies to string backed option sets.
+        /// All enums can be coerced to strings and concatenated as strings.  In some situations, we intend the maker
+        /// to concatenate enums and not lose the strong typing of that enum, for example with JSONFormat or MatchOptions.
+        /// A concatenate with like enum values results in the enum type being preserved if this flag is true.
+        /// </summary>
+        bool CanConcatenateStronglyTyped { get; }
+
+        /// <summary>
+        /// Only applies to number backed option sets.
```

Comment:
>Only applies to number backed option sets. [](http://example.com/codeflow?start=12&length=42)

We should have checks that enforce that (possibly when adding a new option set to the symbol table)

Reply:
I don't know of a way to enforce this at the interface level.  Having a seperate OptionSetDKind is possible, but gets messy considering DKind is a part of DType.  This is enforced in EnumSymbol through a Contracts.Assert.  OptionSet.cs only supports creating String backed option sets.